test/StoragelessTest/Session/Zend/SessionAdapterTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+declare(strict_types=1);
+
+namespace PSR7SessionsTest\Storageless\Session\Zend;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use PSR7Sessions\Storageless\Session\SessionInterface;
+use PSR7Sessions\Storageless\Session\Zend\SessionAdapter;
+
+/**
+ * @covers \PSR7Sessions\Storageless\Session\Zend\SessionAdapter
+ */
+final class SessionAdapterTest extends TestCase
+{
+    public function testToArray() : void
+    {
+        $object      = new \stdClass();
+        $object->key = 'value';
+
+        /** @var SessionInterface|MockObject $session */
```

Comment:
Note: use `&` instead of `|` for new mocks: mocks are intersection types, not union types. Tooling started supporting this synopsis very recently

Reply:
i don't get that, do you have a correct example?

Reply:
```suggestion
        /** @var SessionInterface&MockObject $session */
```

Reply:
if i try that i get the following message:
```
23 | ERROR | [x] Type PHPUnit\Framework\MockObject\MockObject is not used in this file.
```

Reply:
Hmm, I think an upgrade of phpcs and doctrine/coding-style is needed for that to work